iot/iot.rb

```diff
@@ -0,0 +1,795 @@
+  # [END iot_patch_es_device]
+end
+
+
+$patch_rsa_device = -> (project_id:, location_id:, registry_id:, device_id:, cert_path:) do
+  # [START iot_patch_rsa_device]
+  # project_id  = "Your Google Cloud project ID"
+  # location_id = "The Cloud region the registry is located in"
+  # registry_id = "The registry to create a device in"
+  # device_id = "The identifier of the device to patch"
+  # cert_path = "The path to the RSA certificate"
+
+  require "google/apis/cloudiot_v1"
+
+  # Initialize the client and authenticate with the specified scope
+  Cloudiot = Google::Apis::CloudiotV1
+  iot_client = Cloudiot::CloudIotService.new
+  iot_client.authorization = Google::Auth.get_application_default(
+    "https://www.googleapis.com/auth/cloud-platform"
+  )
+
+  # The resource name of the location associated with the project
+  parent = "projects/#{project_id}/locations/#{location_id}/registries/#{registry_id}"
+  path = "#{parent}/devices/#{device_id}"
+
+  credential = Google::Apis::CloudiotV1::DeviceCredential.new
+  credential.public_key = Google::Apis::CloudiotV1::PublicKeyCredential.new
+  credential.public_key.format = "RSA_X509_PEM"
+  credential.public_key.key = File.read(cert_path)
+
+  device = Cloudiot::Device.new
+  device.credentials = [credential]
+
+  # Create the device
+  device = iot_client.patch_project_location_registry_device(
+    path, device, update_mask: 'credentials'
+  )
+
+  puts "Device: #{device.id}"
+  puts "\tBlocked: #{device.blocked}"
+  puts "\tLast Event Time: #{device.last_event_time}"
+  puts "\tLast State Time: #{device.last_state_time}"
+  puts "\tName: #{device.name}"
+  puts "\tCertificate formats:"
+  if device.credentials == nil
+    puts "\t\tNo certificates for device"
+  else
+    device.credentials.each { |cert| puts "\t\t#{cert.public_key.format}"}
+  end
+  # [END iot_patch_rsa_device]
+end
+
+
+$send_device_config = -> (project_id:, location_id:, registry_id:, device_id:, data:) do
+  # [START send_device_config]
+  # project_id  = "Your Google Cloud project ID"
+  # location_id = "The Cloud region the registry is located in"
+  # registry_id = "The registry to get a device from"
+  # device_id = "The identifier of the device to set configurations on"
+  # data = "The data, e.g. {fan: on} to send to the device"
+
+  require "google/apis/cloudiot_v1"
+
+  # Initialize the client and authenticate with the specified scope
+  Cloudiot = Google::Apis::CloudiotV1
+  iot_client = Cloudiot::CloudIotService.new
+  iot_client.authorization = Google::Auth.get_application_default(
+    "https://www.googleapis.com/auth/cloud-platform"
+  )
+
+  # The resource name of the location associated with the project
+  parent = "projects/#{project_id}/locations/#{location_id}"
+  resource = "#{parent}/registries/#{registry_id}/devices/#{device_id}"
+
+  config = Cloudiot::DeviceConfig.new
+  config.binary_data = data
+
+  # Set configuration for the provided device
+  res = iot_client.modify_cloud_to_device_config(
```

Comment:
the `res` variable (result of calling `modify_cloud_to_device_config`) is unused

Is there anything useful in it to print out? If not, call the method without capturing the response in a variable

Reply:
It just has the ETAG and I was not seeing it, going ahead and removing